### PR TITLE
Relaxed typing requirements for cli_utils.print_table

### PR DESCRIPTION
**Pull request by AlexejPenner**
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe changes
Typing restrictions were a bit to tight.

### CODE REVIEW

**Review by htahir1 — CHANGES_REQUESTED** [review]
:-)

File: src/zenml/cli/utils.py [hunk 1]
```diff
@@ -115,14 +115,14 @@ def pretty_print(obj: Any) -> None:
     click.echo(str(obj))
 
 
-def print_table(obj: List[Dict[str, str]]) -> None:
+def print_table(obj: List[Dict[str, Any]]) -> None:
```
Comment (htahir1) [inline src/zenml/cli/utils.py:118]: The tabulate docs say:

```
he following tabular data types are supported:

list of lists or another iterable of iterables
list or another iterable of dicts (keys as columns)
dict of iterables (keys as columns)
two-dimensional NumPy array
NumPy record arrays (names as columns)
pandas.DataFrame
```
Then maybe we can change this to `List[Dict[str, Iterable]])`

Comment (AlexejPenner) [reply src/zenml/cli/utils.py:118]: Here we are in the case of
-> list or another iterable of dicts (keys as columns)
On the level of the dict values we just expect anything that has a string representation I think.

Comment (AlexejPenner) [reply src/zenml/cli/utils.py:118]: in our case we get str or objects as dict values, I could change this to `List[Dict[str, Any[str, object]]` if you prefer

**Pull request merged**